R/app-driver-initialize.R

```diff
@@ -164,6 +192,27 @@ app_initialize_ <- function(
 app_initialize <- function(self, private, ..., view = missing_arg()) {
   ckm8_assert_app_driver(self, private)
 
+  # Always skip on CRAN
+  # https://rstudio.github.io/chromote/articles/example-cran-tests.html
+  #
+  # > Note that the manual says that external commands (here 'chrome') must
+  # > be used conditionally, so changes should not have resulted in a check
+  # > failure.
+  # - Ripley
+  #
+  # However, we can not use it conditionally.
+  # So the only course of action is to skip the test.
+
+  # If sys envvar `SHINYTEST2_APP_DRIVER_TEST_ON_CRAN` is "true" or "1",
+  # then do not skip testing on CRAN
+  env_test_on_cran <- Sys.getenv("SHINYTEST2_APP_DRIVER_TEST_ON_CRAN", "false")
+  should_test_on_cran <- tolower(env_test_on_cran) %in% c("true", "1")
+  should_skip_on_cran <- !should_test_on_cran
+  if (should_skip_on_cran) {
+    # 99% Common case: Skip test on CRAN where chromote is used
+    testthat::skip_on_cran()
+  }
```

Comment:
You can skip the one intermediate variable, I think it actually makes it a little less understandable.

```suggestion
  should_test_on_cran <- tolower(env_test_on_cran) %in% c("true", "1")
  if (!should_test_on_cran) {
    # 99% Common case: Skip test on CRAN where chromote is used
    testthat::skip_on_cran()
  }
```

or the other way

```suggestion
  should_skip_on_cran <- !tolower(env_test_on_cran) %in% c("true", "1")
  if (should_skip_on_cran) {
    # 99% Common case: Skip test on CRAN where chromote is used
    testthat::skip_on_cran()
  }
```